db/util_testing.go

```diff
@@ -0,0 +1,78 @@
+package db
+
+import (
+	"math"
+
+	"github.com/couchbase/gocb"
+	"github.com/couchbase/sync_gateway/base"
+)
+
+// Workaround SG #3570 by doing a polling loop until the star channel query returns 0 results.
+// Uses the star channel index as a proxy to indicate that _all_ indexes are empty (which might not be true)
+func WaitForIndexEmpty(bucket *base.CouchbaseBucketGoCB, bucketSpec base.BucketSpec) error {
```

Comment:
Having a `useXattrs bool` parameter instead of bucketSpec would be more transparent and potentially more flexible/easier to use.

Reply:
Fixed in 000708f6cd889a112c3fd6e06c49955301ff3297